### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by scarroll32**
- Improve clarity on workflow file
- Delete old Travis file
- Update Gemfile matrix versions and delete old ones
- Support for Rails 6, 6.1, 7 and edge

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -27,6 +32,7 @@ jobs:
       with:
         ruby-version: ${{ matrix.ruby_version }}
         bundler-cache: true
+    - name: Install dependencies
+      run: bundle install
```
Comment (simi) [inline .github/workflows/test.yml:36]: this is not needed since `bundler-cache: true` on previous step

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -2,8 +2,6 @@ name: Tests
 on:
   pull_request:
   push:
-    branches:
```
Comment (simi) [inline .github/workflows/test.yml:5]: All PRs are build and pushes to `master` as well. What's the reason to build CI for all pushes? :thinking:

File: .github/workflows/test.yml (hunk 1 above)
Comment (scarroll32) [reply .github/workflows/test.yml]: 👍

File: .github/workflows/test.yml (hunk 2 above)
Comment (scarroll32) [reply .github/workflows/test.yml]: 🆗

**Comment by kerrizor** [comment]
Once we can get checks ✅ I think this would be good to get into a 1.8.0 release (to mark the dropping of explicit support for Rails 4 and 5.)

**Comment by simi** [comment]
Just to make it clear, changing from ruby `3.1` to `3.1.2` it will not use latest 3.1 Ruby anymore which could be beneficial (to not need update CI on each minor update). But it heavily depends on maintainer's preferences. :sun_with_face:

**Comment by scarroll32** [comment]
@kerrizor do the workflows need to be approved on each run?

**Comment by kerrizor** [comment]
> @kerrizor do the workflows need to be approved on each run?
> 
> Actually I found a way to run them locally https://github.com/nektos/act

Thank goodness :/

(I'm biased, as a Gitlabber :D )

**Comment by scarroll32** [comment]
> > @kerrizor do the workflows need to be approved on each run?
> > Actually I found a way to run them locally https://github.com/nektos/act
> 
> Thank goodness :/
> 
> (I'm biased, as a Gitlabber :D )

😉

**Comment by simi** [comment]
> @kerrizor do the workflows need to be approved on each run?

Depends on repo settings, usually only on first contribution.

**Comment by kerrizor** [comment]
@simi  @scarroll32 either of you able to spot the issue that's causing the checks to fail here?

**Comment by scarroll32** [comment]
@kerrizor I can't I'm afraid. The https://github.com/nektos/act solution didn't really work, so I will need to build up the PR commit by commit with a green pipeline on each step.

In order to do that I need to get something accepted, so have shifted gears to https://github.com/voormedia/rails-erd/pull/390/files

I'll mark this as a draft for now so it stops triggering the CI.

**Comment by kerrizor** [comment]
Weirdly... I guess I don't understand Github anymore, because I forked this branch and made a new PR with MY branch, but that triggered the checks to run in this PR as well as on mine? 🤡 

> Just to make it clear, changing from ruby `3.1` to `3.1.2` it will not use latest 3.1 Ruby anymore which could be beneficial (to not need update CI on each minor update). But it heavily depends on maintainer's preferences. 🌞

@scarroll32 I'm slightly in favor of pinning only to the most recent minor release, since Ruby follows pretty good semver hygiene.. do you mind making that change and pushing that up?

**Pull request closed without merging**